pg_rational.c

```diff
@@ -464,6 +470,81 @@ rational_larger(PG_FUNCTION_ARGS)
  ************** INTERNAL ***************
  */
 
+/*
+limit_denomintor() uses continued fractions to convert the
+rational n/d into the rational n'/d' with d' < max_denominator
+and n' <= INT32_MAX and smallest |d/n-d'/n'|.
```

Comment:
Is there a paper or a description of this algorithm you can link to in the comment?